main.py

```diff
@@ -0,0 +1,123 @@
+GITLAB_SERVER = os.environ['GITLAB_SERVER']
+GITLAB_TOKEN = os.environ['GITLAB_TOKEN']
+
+APP_INTERFACE_URL = os.environ['APP_INTERFACE_URL']
+APP_INTERFACE_TOKEN = os.environ['APP_INTERFACE_TOKEN']
+
+SENTRY_DSN = os.getenv('SENTRY_DSN')
+
+gh_client = Github(GITHUB_TOKEN)
+gl_client = Gitlab(GITLAB_SERVER, private_token=GITLAB_TOKEN)
+gql_client = GqlApi(APP_INTERFACE_URL, APP_INTERFACE_TOKEN)
+
+if os.getenv('SLEEP_TIME'):
+    SLEEP_TIME = int(os.environ['SLEEP_TIME'])
+else:
+    SLEEP_TIME = 300
+
+
+if SENTRY_DSN:
+    def before_breadcrumb(crumb, hint):
+        # remove authentication from http urls
+        if crumb['category'] == 'subprocess':
+            crumb['message'] = re.sub(r'//.*?@', '//***@', crumb['message'])
+        return crumb
+
+    sentry_sdk_init(SENTRY_DSN, before_breadcrumb=before_breadcrumb)
+
+
+def inject_auth(url, auth):
+    parts = list(urllib.parse.urlsplit(url))
+    parts[1] = auth + '@' + parts[1]
+    return urllib.parse.urlunsplit(parts)
+
+
+def get_stats(saas_repos):
+    stats = []
+    for saas_repo_full in saas_repos:
+        logging.info(['saas_repo', saas_repo_full])
+
+        if saas_repo_full.startswith(GITLAB_SERVER):
+            saas_repo_auth = inject_auth(saas_repo_full, GITLAB_TOKEN)
+        else:
+            saas_repo_auth = saas_repo_full
+
+        try:
+            saas_repo = SaasRepo(saas_repo_auth)
+        except Exception as e:
+            logging.error(e)
+            continue
+
+        for service in saas_repo.services:
+            upstream_url = service['url']
+            context = service['context']
+            name = service['name']
+            sha = service['hash']
+
+            if sha in ['none', 'ignore', None]:
+                logging.info(['ignore', saas_repo_full, upstream_url,
+                              context, name, sha])
+                continue
+
+            logging.info(['fetching_stats', context, name])
+
+            if upstream_url.startswith(GHRepo.PREFIX):
+                repo = GHRepo(gh_client, upstream_url)
+            elif upstream_url.startswith(gl_client.url):
+                repo = GLRepo(gl_client, upstream_url)
+            else:
+                logging.error(['Invalid upstream repo',
+                               saas_repo_full, upstream_url])
+
+            try:
+                commit_index, commit = repo.get_commit(sha)
+            except CommitNotFound:
+                logging.error(['CommitNotFound', saas_repo_full, upstream_url,
+                               context, name, sha])
+                continue
+
+            stats.append({
+                'context': context,
+                'service': name,
+                'saas_upstream_commits': repo.total_commits,
+                'saas_commit_index': repo.total_commits - commit_index,
+                'saas_commit_ts': repo.commit_ts(commit)
+            })
+
+    return stats
+
+
+if __name__ == "__main__":
+    start_http_server(8000)
```

Comment:
I took a look at this `start_http_server` implementation... I think it's safe to use it here. The service will not serve anything but the `/metrics` endpoint, so no need to go fancy. If it becomes a problem (which I don't think it will), we figure a way out.
Btw, using the `start_http_server` as the way to run this app means that the host and the port should could also come from the command line, but again, keep it simple. This should do to spin the server up:
```
$ python .../.../app.py 0.0.0.0 8000
```